### PR TITLE
Initial PR for System.CommandLine

### PR DESCRIPTION
**Pull request by DavidKarlas**

### CODE REVIEW

File: src/Microsoft.TemplateEngine.Cli/Commands/GlobalArgs.cs [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+        {
+            Quiet = parseResult.ValueForOption(quietOption);
+            DebugCustomSettingsLocation = parseResult.ValueForOption(debugCustomSettingsLocationOption);
+            DebugVirtualizeSettings = parseResult.ValueForOption(debugVirtualizeSettingsOption);
+            DebugAttach = parseResult.ValueForOption(debugAttachOption);
+            DebugReinit = parseResult.ValueForOption(debugReinitOption);
+            DebugRebuildCache = parseResult.ValueForOption(debugRebuildCacheOption);
+            DebugShowConfig = parseResult.ValueForOption(debugShowConfigOption);
+            //TODO: check if it gets the command name correctly.
+            CommandName = parseResult.CommandResult.Command.Name;
```
Comment (vlada-shubina) [inline src/Microsoft.TemplateEngine.Cli/Commands/GlobalArgs.cs:55]: @DavidKarlas @bekir-ozturk cannot find good solution for this for now. 
`parseResult.CommandResult.Command.Name` is name of the last command here, whereas we need to now the name of `NewCommand` which can be `new` or `new3`.  

Any ideas? 
Don't really want to hardcode to use parent, as if we implement alias operations with sub command `new` might be different level.

File: src/Microsoft.TemplateEngine.Cli/Commands/BaseCommand.cs [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+        }
+
+        protected virtual IEnumerable<string> GetSuggestions(TArgs args, IEngineEnvironmentSettings environmentSettings, string? textToMatch)
+        {
+            return base.GetSuggestions(args.ParseResult, textToMatch);
+        }
+
+        protected IEngineEnvironmentSettings CreateEnvironmentSettings(TArgs args)
+        {
+            string? outputPath = (args as InstantiateCommandArgs)?.OutputPath;
```
Comment (vlada-shubina) [inline src/Microsoft.TemplateEngine.Cli/Commands/BaseCommand.cs:108]: can we do any better? that's already broken

- reparse it inline to get output path?
- add output path as global option?
- recreate env settings before instantiating?

Comment (DavidKarlas) [reply src/Microsoft.TemplateEngine.Cli/Commands/BaseCommand.cs:108]: Feels more like there is problem with `GlobalJsonFileExistsInPath` or `TryGetHostParamDefault`...
1st `GlobalJsonFileExistsInPath` should probably be Macro that takes Output folder as input.
2nd Maybe `TryGetHostParamDefault` needs access to existing Symbols/Output

File: src/Microsoft.TemplateEngine.Cli/Commands/TemplateGroupArgs.cs [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+using System.CommandLine.Parsing;
+using Microsoft.TemplateEngine.Abstractions;
+
+namespace Microsoft.TemplateEngine.Cli.Commands
+{
+    internal class TemplateGroupArgs
+    {
+        private static Option<string> _outputOption = new Option<string>("-o");
+        private static Option<string> _nameOption = new Option<string>("-n");
+        private static Option<bool> _helpOption = new Option<bool>(new string[] { "-h", "--help", "-?" });
```
Comment (vlada-shubina) [inline src/Microsoft.TemplateEngine.Cli/Commands/TemplateGroupArgs.cs:16]: note: not all options are defined

File: src/Microsoft.TemplateEngine.Cli/Commands/TemplateGroupCommand.cs [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+        public TemplateGroupCommand(NewCommand newCommand, IEngineEnvironmentSettings environmentSettings, ITemplateInfo template) : base(template.ShortNameList[0], template.Name + Environment.NewLine + template.Description)
+        {
+            _newCommand = newCommand;
+            _environmentSettings = environmentSettings;
+            foreach (var item in template.ShortNameList.Skip(1))
+            {
+                AddAlias(item);
+            }
+            _templateSpecificOptions = TemplateGroupArgs.AddToCommand(this, template);
```
Comment (vlada-shubina) [inline src/Microsoft.TemplateEngine.Cli/Commands/TemplateGroupCommand.cs:26]: probably we want to move this logic to command itself, as with other Args classes

File: test/Microsoft.TemplateEngine.Cli.UnitTests/TabCompletionTests.cs [hunk 5]
```diff
@@ -0,0 +1,28 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.CommandLine;
+using Microsoft.TemplateEngine.TestHelper;
+using Xunit;
+
+namespace Microsoft.TemplateEngine.Cli.UnitTests
+{
+    public class TabCompletionTests
```
Comment (vlada-shubina) [inline test/Microsoft.TemplateEngine.Cli.UnitTests/TabCompletionTests.cs:10]: this has to be fixed

**Review by vlada-shubina — APPROVED** [review]
This PR is not complete, but imo good to go for now.

**Pull request merged**